src/agent/agent.c

```diff
@@ -135,11 +135,47 @@ static int agent_disconnected(UNUSED sd_bus_message *message, void *userdata, UN
 
 static int agent_reset_heartbeat_timer(Agent *agent, sd_event_source **event_source);
 
+static int agent_check_controller_liveness(Agent *agent) {
```

Comment:
Since this is returning only two values, lets use a `bool` here instead.

Reply:
done